### ISSUE TITLE
local tests not running on darwin

### ISSUE THREAD

**Issue by PeerHerholz** [nipype/pydra#293]
Due to `Singularity` [differences between linux and darwin](https://sylabs.io/guides/3.3/user-guide/installation.html#install-on-windows-or-mac), the local tests outlined in the [contributing guidelines](https://github.com/nipype/pydra/blob/master/CONTRIBUTING.md) are failing.

After a quick chat about it, we have two options:

- [ ] check differences between linux and darwin, including an `ENV` that adapts the tests for darwin accordingly
- [ ] check OS upon running the tests to provide contributors with a corresponding message that tests don't work on darwin

**Comment by djarecka** [comment]
@PeerHerholz @satra - do you think it's worth it to test the OSX Singularity? Originally I've decided to ignore it.

**Comment by PeerHerholz** [comment]
Puh, good question. I think that's more for you developers to decide. IMHO it would come down to 1) how many contributors across what experience levels are expected and 2) are the tests and/or `Singularity` behavior on darwin (and maybe windows) expected to change?
Depending on the approach, both options mentioned above should be ok I guess, given that a behavior such as "if tests fail locally based on your OS you can still submit a PR" will be allowed. I would of course be happy if tests would work on darwin as I do my day to day coding work on that, but that's just me. Anyway, I would vote for definitely including the pointer re OS incompatibilities somewhere.

**Comment by djarecka** [comment]
yes, we definitely should be checking the system, but I'd personally just vote for adding `pytest.mark.skip` for OS. But wanna see if there is any opposition to that

**Comment by satra** [comment]
i think its not just skipping the test, if someone tries to run singularity on osx, we should raise an error in pydra if doesn't support singularity on osx. @djarecka - can you or nicole check what the limitations are why we cannot run it?

**Comment by satra** [comment]
or @PeerHerholz

**Comment by djarecka** [comment]
@PeerHerholz - let me know if this is something that you would be interested in checking

**Comment by PeerHerholz** [comment]
Since I started it... hehe, jk. Sure thing, I'll have a look!

**Comment by PeerHerholz** [comment]
Hi gang,

I tried a few things after downloading the latest [Singularity desktop beta](https://sylabs.io/singularity-desktop-macos/). The errors I get during the tests are `assertion errors`, I do however get the output message in my shell (e.g. `hello from pydra`). I'm not sure, but I think one problem might be the `/tmp` directory as I had some issues with that before after recent changes introduced in `Catalina` wrt what/who can access what where. I attached the log of the tests: 
[pydra_test_darwin.txt](https://github.com/nipype/pydra/files/4839224/pydra_test_darwin.txt)

**Comment by djarecka** [comment]
@PeerHerholz - thank you for checking. Perhaps I should download singularity, I'm not sure what is wrong...
